packages/location-management/src/components/LocationUnitGroupAddEdit/Form.tsx

```diff
@@ -0,0 +1,169 @@
+import { Ripple } from '@onaio/loaders';
+
+const layout = { labelCol: { span: 8 }, wrapperCol: { span: 11 } };
+const offsetLayout = { wrapperCol: { offset: 8, span: 11 } };
+
+const status = [
+  { label: 'Active', value: true },
+  { label: 'Inactive', value: false },
+];
+
+interface FormField {
+  name: string;
+  description: string;
+  active: boolean;
+}
+
+/** yup validations for practitioner data object from form */
+const userSchema = Yup.object().shape({
+  name: Yup.string().typeError('Name must be a String').required('Name is Required'),
+  active: Yup.boolean().typeError('Status must be a Boolean').required('Status is Required'),
+  description: Yup.string()
+    .typeError('Description must be a String')
+    .required('Description is Required'),
+});
+
+interface Props {
+  id?: string;
+}
+
+/**
+ * Handle form submission
+ *
+ * @param {Object} values the form fields
+ * @param {string} accessToken api access token
+ * @param {object} props component props
+ * @param {Function} setSubmitting method to set submission status
+ */
+export const onSubmit = (
+  values: FormField,
+  accessToken: string,
+  props: Props,
+  setSubmitting: (isSubmitting: boolean) => void
+) => {
+  const serve = new OpenSRPService(accessToken, API_BASE_URL, LOCATION_UNIT_GROUP_ALL);
+
+  const payload: LocationUnitGroupPayloadPOST | LocationUnitGroupPayloadPUT = values;
+
+  if (props.id) {
+    (payload as LocationUnitGroupPayloadPUT).id = props.id;
+    serve
+      .update(payload)
+      .then(() => {
+        sendSuccessNotification('Location Unit Group updated successfully');
+        setSubmitting(false);
+        history.goBack();
+      })
+      .catch(() => {
+        sendErrorNotification('An error occurred');
+        setSubmitting(false);
+      });
+  } else {
+    serve
+      .create(payload)
+      .then(() => {
+        sendSuccessNotification('Location Unit Group successfully');
+        setSubmitting(false);
+        history.goBack();
+      })
+      .catch(() => {
+        sendErrorNotification('An error occurred');
+        setSubmitting(false);
+      });
+  }
+};
+
+export const Form: React.FC<Props> = (props: Props) => {
+  const accessToken = useSelector((state) => getAccessToken(state) as string);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [initialValue, setInitialValue] = useState<FormField>({
+    name: '',
+    description: '',
+    active: true,
+  });
+
+  useEffect(() => {
+    if (isLoading) {
+      if (props.id) {
+        const serve = new OpenSRPService(
+          accessToken,
+          API_BASE_URL,
+          LOCATION_UNIT_GROUP_GET + props.id
+        );
+        serve
+          .list()
+          .then((response: LocationUnitGroup) => {
+            setInitialValue({
+              active: response.active,
+              description: response.description,
+              name: response.name,
+            });
+            setIsLoading(false);
```

Comment:
Maybe we should have `setIsLocading(false)` in the finally block? If there is an error, the loader will still show, is it the behaviour we want?

Reply:
i dont think so though i am not sure ..... if there is some error it will show the error notification without loading an error .. maybe we can use sanitary or notification module to deal with error like these by redirecting or something ?

Reply:
@ashfahan In other components, we are also setting isLoading to false in the catch block, I was thinking for consistency we should also do the same here

Reply:
@dubdabasoduba should we create a new issue for this and solve it after conversation ?